codec_union.go

```diff
@@ -303,6 +303,10 @@ func (d *unionResolvedDecoder) Decode(ptr unsafe.Pointer, r *Reader) {
 }
 
 func unionResolutionName(schema Schema) string {
+	if name := getLogicalTypeName(schema); name != "" {
```

Comment:
I am concerned that this will trump other named registered types. Perhaps instead of a new function, add it as a fallback to `schemaTypeName`. ie we take the full name or the type name plus any logical postfix. WDYT?

Reply:
That's a good point. I'll make the change and unregister `fixed.decimal` and `fixed.duration` because those would be named types.